odl/solvers/functional/example_functionals.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Example functions used in optimization."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.solvers.functional.functional import Functional
+from odl.operator.operator import Operator
+from odl.space.base_ntuples import FnBase
+from odl.space.npy_ntuples import MatVecOperator
+
+
+__all__ = ('RosenbrockFunctional',)
+
+
+class RosenbrockFunctional(Functional):
+
+    """The famous Rosenbrock funciton on ``R^n``.
```

Comment:
Perhaps we should, but the first one is not feasible for a general implementation since it only works for spaces of even dimension. I'll write this in the notes